node/actors/executor/src/lib.rs

```diff
@@ -20,14 +19,15 @@ mod tests;
 pub use self::config::{proto, ConsensusConfig, ExecutorConfig, GossipConfig};
 
 /// Validator-related part of [`Executor`].
-#[derive(Debug)]
 struct ValidatorExecutor {
     /// Consensus network configuration.
     config: ConsensusConfig,
     /// Validator key.
     key: validator::SecretKey,
     /// Store for replica state.
     replica_state_store: Arc<dyn ReplicaStateStore>,
+    /// Payload proposer for new blocks.
+    payload_source: Arc<dyn PayloadSource>,
```

Comment:
Is there a particular reason you didn't include `fmt::Debug` as a supertrait for `PayloadSource`? It would allow to retain `Debug` impls for `ValidatorExecutor` / `Executor` and not change `unwrap_err()` to `err().unwrap()` in tests. I'm not sure to which extent we'll copy our code style for published crates from the standard library, but I'd strongly advocate for [debuggability](https://rust-lang.github.io/api-guidelines/debuggability.html#all-public-types-implement-debug-c-debug) (all public types implementing `Debug`) in any case.

Reply:
> I don't see how unwrap_err() is better than err().unwrap()

It allows to inspect the `Ok` variant contents, which improves debuggability. Likewise, having a fuller view into `ValidatorExecutor` improves debuggability as well (which is not public, but is present as a field in a public `Executor` type). IMO, having an `fmt::Debug` constraint is not a burden; and it's already a supertrait for adjacent storage traits. OTOH, that's a minor thing, and I'm not in position to insist 🤷

Reply:
ack